src/work/ConditionalWork.h

```diff
@@ -8,8 +8,51 @@
 namespace stellar
 {
 
+// Any function used as a ConditionFn should be monotonic: return false for the
+// first 0 or more times it's called, then if ever returns true, always return
+// true thereafter. It will be called repeatedly until it returns true, but
+// never again (it will actually be deleted once it returns true once). No more
+// detailed reset/retry protocol exists in the relationship between
```

Comment:
I don't understand what this last sentence means.. do you mean that conditional work doesn't retry?

Reply:
No, I mean that the ConditionalWork doesn't propagate any of its lifecycle transitions into the ConditionFn. I'll rephrase.